File: crates/warfarin-core/src/models/outcome.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WfOutcome {
  pub id: i64,
  pub hn: String,
  pub event_date: String,
  pub event_type: String,
  pub description: Option<String>,
  pub inr_at_event: Option<f64>,
  pub action_taken: Option<String>,
  pub created_by: Option<String>,
  pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct OutcomeInput {
  pub hn: String,
  pub event_date: String,
  pub event_type: String,
  pub description: Option<String>,
  pub inr_at_event: Option<f64>,
  pub action_taken: Option<String>,
  pub created_by: Option<String>,
}

File: src/types/outcome.ts
export type OutcomeType =
  | 'major_bleeding'
  | 'minor_bleeding'
  | 'thromboembolism'
  | 'hospitalization'
  | 'death'
  | 'other';

export interface WfOutcome {
  id: number;
  hn: string;
  eventDate: string;
  eventType: OutcomeType;
  description?: string;
  inrAtEvent?: number;
  actionTaken?: string;
  createdBy?: string;
  createdAt: string;
}

export interface OutcomeInput {
  hn: string;
  eventDate: string;
  eventType: OutcomeType;
  description?: string;
  inrAtEvent?: number;
  actionTaken?: string;
  createdBy?: string;
}
